feat(GDB-10456) Set up GraphQL Playground Component project

teodossidossev opened the pull request:
## WHAT:
- Added new graphql-playground-component.
- Updated dependencies in package.json
- Configured stencil.config.ts to:
  - Support Angular integration.
  - Include plugins like sass and Rollup plugins (commonjs, node-resolve). 
  - Handle asset copying for GraphiQL and related libraries. 
  - Added development notes in DEVELOPMENT.md for project setup, workflow, and integration.

## WHY:
To provide a GraphQL playground component using StencilJS for integration with other projects.

## HOW:
- Updated StencilJS configuration for modularity and flexibility:
- Integrated Angular output target for Angular projects.
- Added support for SCSS with the sass plugin.
- Added functionality for dynamically loading GraphiQL assets (JS, CSS) and rendering the GraphQL playground using React and GraphiQL libraries.
- Documented the development workflow, including submodule handling, asset management, and publishing to npm.

teodossidossev committed:
feat(GDB-10456) Set up GraphQL Playground Component project

## WHAT:
- Added new graphql-playground-component.
- Updated dependencies in package.json
- Configured stencil.config.ts to:
-- Support Angular integration.
-- Include plugins like sass and Rollup plugins (commonjs, node-resolve).
-- Handle asset copying for GraphiQL and related libraries.
-- Added development notes in DEVELOPMENT.md for project setup, workflow, and integration.

## WHY:
To provide a GraphQL playground component using StencilJS for integration with other projects.

## HOW:
- Updated StencilJS configuration for modularity and flexibility:
- Integrated Angular output target for Angular projects.
- Added support for SCSS with the sass plugin.
- Added functionality for dynamically loading GraphiQL assets (JS, CSS) and rendering the GraphQL playground using React and GraphiQL libraries.
- Documented the development workflow, including submodule handling, asset management, and publishing to npm.

svilenvelikov commented on ontotext-graphql-playground-component/src/components/assets/explorer.index.umd.js:
I'm pretty sure that these should not be commited

teodossidossev committed:
feat: Add Cypress integration, update Stencil dependencies, and enhance GraphQL Playground component

- Integrated Cypress for E2E testing:
  - Added Cypress configuration, commands, and steps.
  - Configured `cypress.config.js` and `cypress.tsconfig.json`.

- Updated Stencil project configuration:
  - Added a custom dev server middleware in `stencil.config.ts`.
  - Introduced Angular output target enhancements.

- Enhanced GraphQL Playground component:
  - Refined asset loading with `getAssetPath`.
  - Added `dev-server.js` for custom middleware setup.
  - Improved default view setup with updated HTML templates.

- Dependency Updates:
  - Moved some dependencies to `devDependencies` for better build optimization.
  - Added `rxjs` and updated TypeScript to `5.6.3`.

- Miscellaneous:
  - Added comments and documentation for custom middleware and Cypress commands.
  - Resolved build warnings and adjusted dev server request listener path.

svilenvelikov commented on DEVELOPMENT.md:
```diff
@@ -0,0 +1,162 @@
+    - Test thoroughly after any updates to ensure no breaking changes affect the StencilJS component.
+
+2. **Code Consistency**
+    - Follow the coding standards of the Ontotext organization for both the `graphiql` submodule and the StencilJS component.
```
Development in the `graphiql` must comply with the practices and standarts in that project. This is needed to prevent spreading unwanted changes in the codebase hance preventing conflicts during upgrade.

teodossidossev replied:
done

svilenvelikov commented on DEVELOPMENT.md:
````diff
@@ -0,0 +1,162 @@
+  ```
+- Update the StencilJS component to wrap `graphiql` features with Ontotext-specific styling and behavior.
+
+### 4. **Building and Publishing**
````
This is expected to be automated in jenkins. Right?

teodossidossev replied:
yes

teodossidossev committed:
added lint, fix tests commands

teodossidossev committed:
refined DEVELOPMENT.md text
